src/engine/bank05.asm

```diff
@@ -320,4 +1514,3421 @@ ZeroData: ; 1575e (5:575e)
+CheckIfActivePokemonCanUseAnyNonResidualMove: ; 162a1 (5:62a1)
+	xor a ; active card
+	ldh [hTempPlayAreaLocation_ff9d], a
+; first move
+	ld [wSelectedMoveIndex], a
+	call CheckIfCardCanUseSelectedMove
+	jr c, .next_move
+	ld a, [wLoadedMoveCategory]
+	and RESIDUAL
+	jr z, .ok
+
+.next_move
+; second move
+	ld a, $01
+	ld [wSelectedMoveIndex], a
+	call CheckIfCardCanUseSelectedMove
+	jr c, .fail
+	ld a, [wLoadedMoveCategory]
+	and RESIDUAL
+	jr z, .ok
+.fail
+	or a
+	ret
+
+.ok
+	scf
+	ret
+; 0x162c8
+
+; looks for energy card(s) in hand depending on
+; what is needed for selected card, for both moves
+; 	- if one basic energy is required, look for that energy;
+;	- if one colorless is required, create a list at wDuelTempList
+;	  of all energy cards;
+;	- if two colorless are required, look for double colorless;
+; return carry if successful in finding card
+; input:
+; 	[hTempPlayAreaLocation_ff9d] = location of Pokémon card
+LookForEnergyNeededInHand: ; 162c8 (5:62c8)
+	xor a ; first move
+	ld [wSelectedMoveIndex], a
+	call CheckEnergyNeededForAttack
+	ld a, b
+	add c
+	cp 1
+	jr z, .one_energy
+	cp 2
+	jr nz, .second_move
+	ld a, c
+	cp 2
+	jr z, .two_colorless
+
+.second_move
+	ld a, $01 ; second move
+	ld [wSelectedMoveIndex], a
+	call CheckEnergyNeededForAttack
+	ld a, b
+	add c
+	cp 1
+	jr z, .one_energy
+	cp 2
+	jr nz, .no_carry
+	ld a, c
+	cp 2
+	jr z, .two_colorless
+.no_carry
+	or a
+	ret
+
+.one_energy
+	ld a, b
+	or a
+	jr z, .one_colorless
+	ld a, e
+	call LookForCardIDInHandList
+	ret c
+	jr .no_carry
+
+.one_colorless
+	call CreateEnergyCardListFromHand
+	jr c, .no_carry
+	scf
+	ret
+
+.two_colorless
+	ld a, DOUBLE_COLORLESS_ENERGY
+	call LookForCardIDInHandList
+	ret c
+	jr .no_carry
+; 0x16311
+
+; looks for energy card(s) in hand depending on
+; what is needed for selected card and move
+; 	- if one basic energy is required, look for that energy;
+;	- if one colorless is required, create a list at wDuelTempList
+;	  of all energy cards;
+;	- if two colorless are required, look for double colorless;
+; return carry if successful in finding card
+; input:
+; 	[hTempPlayAreaLocation_ff9d] = location of Pokémon card
+;	[wSelectedMoveIndex]		 = selected move to examine
+LookForEnergyNeededForMoveInHand: ; 16311 (5:6311)
+	call CheckEnergyNeededForAttack
+	ld a, b
+	add c
+	cp 1
+	jr z, .one_energy
+	cp 2
+	jr nz, .done
+	ld a, c
+	cp 2
+	jr z, .two_colorless
+.done
+	or a
+	ret
+
+.one_energy
+	ld a, b
+	or a
+	jr z, .one_colorless
+	ld a, e
+	call LookForCardIDInHandList
+	ret c
+	jr .done
+
+.one_colorless
+	call CreateEnergyCardListFromHand
+	jr c, .done
+	scf
+	ret
+
+.two_colorless
+	ld a, DOUBLE_COLORLESS_ENERGY
+	call LookForCardIDInHandList
+	ret c
+	jr .done
+; 0x1633f
+
+; goes through $00 terminated list pointed 
+; by wcdae and compares it to each card in hand.
+; Sorts the hand in wDuelTempList so that the found card IDs
+; are in the same order as the list pointed by de.
+SortTempHandByIDList: ; 1633f (5:633f)
```

Comment:
The wording on this one seems weird but without knowing what sets ``wcdae`` it's hard to suggest an alternative... Anyway, I think the "ByID" is redundant since ``wDuelTempList`` always contains a list of card IDs if it doesn't contain moves/pokemon powers. I would maybe use ``SortTempHandCardList`` for now.

Reply:
I meant it "sort it by ID", instead of sorting it by some other thing like card index, if you understand what I mean. "By ID" refers to how it is sorted, not by what it contains. If it's still ambiguous, I can change it to better reflect that functionality.